odo/backends/url.py

```diff
@@ -133,7 +133,9 @@ def append_urlX_to_X(target, source, **kwargs):
         with open(target.path, 'wb') as fp:
             for chunk in iter(curry(r.read, chunk_size), ''):
                 fp.write(chunk)
-                return target
+                if not chunk:
```

Comment:
sorry, I understand that it is returning an empty string, but maybe that means our `iter` is written incorrectly and should be `iter(curry(...), b'')`. This is what two argument iter is supposed to solve.